src/nbl/video/CVulkanLogicalDevice.cpp

```diff
@@ -785,4 +841,78 @@ bool CVulkanLogicalDevice::getQueryPoolResults(IQueryPool* queryPool, uint32_t f
     return ret;
 }
 
+core::smart_refctd_ptr<IGPUBuffer> CVulkanLogicalDevice::createBuffer(IGPUBuffer::SCreationParams&& creationParams)
+{
+ 
+    VkBufferCreateInfo vk_createInfo = { VK_STRUCTURE_TYPE_BUFFER_CREATE_INFO };
+    // Each pNext member of any structure (including this one) in the pNext chain must be either NULL or a pointer to a valid instance of VkBufferDeviceAddressCreateInfoEXT, VkBufferOpaqueCaptureAddressCreateInfo, VkDedicatedAllocationBufferCreateInfoNV, VkExternalMemoryBufferCreateInfo, VkVideoProfileKHR, or VkVideoProfilesKHR
+
+    VkExternalMemoryBufferCreateInfo externalMemoryInfo = {
+       .sType = VK_STRUCTURE_TYPE_EXTERNAL_MEMORY_BUFFER_CREATE_INFO,
+       .handleTypes = creationParams.externalHandleType.value,
+    };
+
+    const bool external = creationParams.externalHandleType.value;
+    const bool imported = external && creationParams.externalHandle;
+    const bool exported = external && !imported;
+
+    vk_createInfo.pNext = external ? &externalMemoryInfo : nullptr;
+    vk_createInfo.flags = static_cast<VkBufferCreateFlags>(0u); // Nabla doesn't support any of these flags
+    vk_createInfo.size = static_cast<VkDeviceSize>(creationParams.size);
+    vk_createInfo.usage = getVkBufferUsageFlagsFromBufferUsageFlags(creationParams.usage);
+    vk_createInfo.sharingMode = creationParams.isConcurrentSharing() ? VK_SHARING_MODE_CONCURRENT : VK_SHARING_MODE_EXCLUSIVE;
+    vk_createInfo.queueFamilyIndexCount = creationParams.queueFamilyIndexCount;
+    vk_createInfo.pQueueFamilyIndices = creationParams.queueFamilyIndices;
+
+    if (external)
+    {
+        VkPhysicalDeviceExternalBufferInfo bufferInfo = {
+            .sType = VK_STRUCTURE_TYPE_PHYSICAL_DEVICE_EXTERNAL_BUFFER_INFO,
+            .usage = vk_createInfo.usage,
+            .handleType = VkExternalMemoryHandleTypeFlagBits(creationParams.externalHandleType.value),
+        };
+
+        VkExternalBufferProperties externalProps = { VK_STRUCTURE_TYPE_EXTERNAL_BUFFER_PROPERTIES };
+        VkPhysicalDevice vk_physicalDevice = static_cast<const CVulkanPhysicalDevice*>(m_physicalDevice)->getInternalObject();
+        vkGetPhysicalDeviceExternalBufferProperties(vk_physicalDevice, &bufferInfo, &externalProps);
```

Comment:
dont query at runtime, query during physical device Limit polling and expose as an `[HANDLE_TYPE_COUNT][USAGES]` array in `SPhysicalDeviceLimits` unless handle type are a bitfield and joining them post-facto is not valid

Reply:
We would have to query each handle type bit for every combination of BuferUsage.

Reply:
yea that'd be dumb, you're right